newsrevenuehub/rev-engine
# Dev-1691 new data source for contributor portal

Please fill out each section even if it's just with "N/A"

#### Plan? (if this draft/incomplete indicate what you intend to do and how)
Modify existing Endpoint to pull contributions data from stripe instead of application database for contributors portal.

#### What's this PR do?
1. Celery tasks will pull contributions data from stripe and loads it to redis cache when ever a magic link is requested.
2. As contributor access the contributor portal, the stripe data from cache is served.
3. Revenue Program is send as query params from contributor portal.
4. All contributions are faceted by revenue program.

#### Why are we doing this? How does it help us?
Contributor will be able to view contributions done before RevEngine went live.

#### How should this be manually tested?
Request for magic link.
Open the contributor portal with the magic link you got.
You should be able to view the contributions faceted by the revenueprogram.

#### Have automated unit tests been added?
Yes.

#### How should this change be communicated to end users?
NA

#### Are there any smells or added technical debt to note?
Celery worker runs when the user requests a magic link, data gets saved in redis for about 30 mins.
If data is not available in cache / expired the celery worker gets triggered again on contributor portal access.
This causes delay to view contributions and user has to keep on refreshing the page for contributions. This problem needs to be addressed in future.

#### Has this been documented? If so, where?
Not yet

#### What are the relevant tickets? Add a link to any relevant ones.
[DEV-1691]
https://news-revenue-hub.atlassian.net/browse/DEV-283
https://news-revenue-hub.atlassian.net/browse/DEV-1567

#### Do any changes need to be made before deployment to production (adding environment variables, for example)?
NA

#### Are there next steps? If so, what? Have tickets been opened for them?
NA


[DEV-1691]: https://news-revenue-hub.atlassian.net/browse/DEV-1691?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
